tests/drivers/opamp/opamp_api/src/main.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/opamp.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+static const enum opamp_gain gain[] = {
+	DT_FOREACH_PROP_ELEM_SEP(DT_PHANDLE(DT_PATH(zephyr_user), opamp),
+				programmable_gain, DT_ENUM_IDX_BY_IDX, (,))
+};
+
+const struct device *get_opamp_device(void)
+{
+	return DEVICE_DT_GET(DT_PHANDLE(DT_PATH(zephyr_user), opamp));
+}
+
+static const struct device *init_opamp(void)
+{
+	int ret;
+
+	const struct device *const opamp_dev =
+				DEVICE_DT_GET(DT_PHANDLE(DT_PATH(zephyr_user), opamp));
+
+	ret = device_is_ready(opamp_dev);
+
+	zassert_true(ret, "OPAMP device %s is not ready", opamp_dev->name);
+
+	return opamp_dev;
+}
+
```

Comment:
```suggestion

ZTEST(opamp, test_ init_opamp)
{
	const struct device *opamp_dev = init_opamp();
	ARG_UNUSED(opamp_dev);
}
```

I know, this test is trivial, but during new driver development it shall be the first test, which have to pass.